api/src/test/java/org/apache/iceberg/expressions/TestAggregateBinding.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.expressions;
+
+import static org.apache.iceberg.expressions.Expression.Operation.COUNT;
+import static org.apache.iceberg.expressions.Expression.Operation.COUNT_STAR;
+import static org.apache.iceberg.expressions.Expression.Operation.MAX;
+import static org.apache.iceberg.expressions.Expression.Operation.MIN;
+import static org.apache.iceberg.expressions.Expressions.max;
+import static org.apache.iceberg.expressions.Expressions.ref;
+import static org.apache.iceberg.types.Types.NestedField.required;
+
+import java.util.Arrays;
+import java.util.List;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.types.Types.StructType;
+import org.assertj.core.api.Assertions;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestAggregateBinding {
+  private static final List<Expression.Operation> AGGREGATES = Arrays.asList(COUNT, MAX, MIN);
+  private static final StructType struct =
+      StructType.of(required(10, "x", Types.IntegerType.get()));
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testAggregateBinding() {
+    for (Expression.Operation op : AGGREGATES) {
+      UnboundAggregate unbound = new UnboundAggregate(op, ref("x"));
```

Comment:
Why not use the `Expressions` factory methods to create unbound expressions?

Reply:
Fixed. Thanks!